app/components/SecureTokenBox.tsx

```diff
@@ -0,0 +1,140 @@
+
+  const setBlockchain = useCallback((event: ChangeEvent<HTMLSelectElement>) => {
+    setBlockchainOption(event.target.value);
+  }, []);
+
+  const secureTokenWrapper = useCallback(async () => {
+    const response = await generateSecureToken({ethAddress, blockchains: showBuyQuoteURLText ? blockchains : [blockchainOption.toLowerCase()]})
+    console.log("generateSecureToken");
+    try {
+      if (response) {setSecureToken(response);} else {setSecureToken('')}
+    } catch (error) {
+      console.log(error)
+      alert(error);
+      console.error(error);
+    }}, [ethAddress,  showBuyQuoteURLText, blockchains, blockchainOption]);
+
+  const linkReady = useMemo(() => secureToken.length > 0, [secureToken]);
+
+  const link = useMemo(() => {
+    if (!linkReady) return "Generate a secure token first to create your one time URL";
+    return (
+      "https://pay.coinbase.com/buy/select-asset?sessionToken=" + secureToken + 
+      (aggregatorInputs?.quoteID ? "&quoteId=" + aggregatorInputs.quoteID : "") +
+      (aggregatorInputs?.defaultAsset ? "&defaultAsset=" + aggregatorInputs.defaultAsset : "") +
+      (aggregatorInputs?.defaultPaymentMethod ? "&defaultPaymentMethod=" + aggregatorInputs.defaultPaymentMethod : "") +
+      (aggregatorInputs?.defaultNetwork ? "&defaultNetwork=" + aggregatorInputs.defaultNetwork : "") +
+      (aggregatorInputs?.fiatCurrency ? "&fiatCurrency=" + aggregatorInputs.fiatCurrency : "") +
+      (aggregatorInputs?.presentFiatAmount ? "&presetFiatAmount=" + aggregatorInputs.presentFiatAmount : "")
+    );
+  }, [linkReady, secureToken, aggregatorInputs]);
+
+  const launch = useCallback(() => {
+    open(link, "_blank", "popup,width=540,height=700")
+  }, [link]);
+
+  const helperText = showBuyQuoteURLText ?
+  <h2> The generated link initializes the Coinbase Onramp URL with the appropriate parameters to execute that buy in just one click for the user. </h2> :
+  <h2>Generate a secure one time URL to launch an Onramp session. Provide a <b>destination wallet address</b> and a supporting <b>blockhain network</b>.</h2>
+
+  const buyQuoteURLDirections = (
+    <div className="flex flex-col ml-10 gap-1 w-2/5">
+      <h2 > 1. Generate a Buy Quote in the section above to get the input parameters to create a secure Onramp URL. </h2>
+      <h2> 2. Enter a <b>destination wallet address</b> and a supporting <b>blockchain network</b> then click <b>&lsquo;Generate secure token&rsquo;</b>. </h2>
+      <h2> 3. Click <b> Launch Onramp </b> to see the one-click buy experience for your users. </h2>
+    </div>
+  )
+
+  return ( 
+    <Card>
+      <div className="flex flex-col p-10 pb-5 gap-1">
+        <h1 className="font-bold underline"> <Link color="foreground" href="https://docs.cdp.coinbase.com/onramp/docs/api-initializing/" isExternal> Generate Secure Onramp Token & URL: </Link> </h1>
+        {helperText}
+      </div>
+      
+
+    
+    {!showBuyQuoteURLText || showBuyQuoteURLText && aggregatorInputs?.quoteID ? 
```

Comment:
Large ternaries hurt readability

```suggestion
    {(!showBuyQuoteURLText || showBuyQuoteURLText && aggregatorInputs?.quoteID) &&